python/lsst/daf/butler/core/datasets/ref.py

```diff
@@ -89,6 +89,29 @@ def _check_component(cls, v: Any, values: Dict[str, Any]) -> Any:  # noqa: N805
             raise ValueError(f"datasetType ({values[d]}) can not be set if component is given ({v}).")
         return v
 
+    @classmethod
+    def direct(cls, *, id: Optional[Union[str, int]] = None, datasetType: Optional[Dict[str, Any]] = None,
+               dataId: Optional[Dict[str, Any]] = None, run: str = None, component=None
+               ) -> SerializedDatasetRef:
+        """Construct a `SerializedDatasetRef` directly without validators.
+
+        This differs from PyDantics construct method in that the arguments are
```

Comment:
```suggestion
        This differs from pydantic construct method in that the arguments are
```
This seems to be the spelling and only capitalization in use in their docs.

I'm also very concerned that we have to add these methods at all; it's bad enough that we have to add `to_simple` and `from_simple` to everything, but this really doesn't seem scale up to multiple levels of composition or normalization.  Why can't we use `construct`?

Reply:
Nate explained to me that this is because `construct` isn't recursive (which I can now see in the docs; it just didn't make sense without a better understanding of what `construct` does).  I can see that we're kind of boxed in; that makes me concerned about pydantic possibly being something that doesn't really meet our needs (which is not to say that I'm aware of anything better), but I can't fault this ticket for that.

Reply:
My motivation for pydantic was butler client/server and fastapi automatically vetting content before it is even passed to butler.